### PR TITLE
Chk 3283 project setup

### PR DESCRIPTION
**Pull request by pietro-tota**
#### List of Changes
Add basic repository structure that includes:
- pull request template
- GHA check pr size action
- Dockerfile (with otel agent for instrumentation)
- Docker compose with service, mongo and mongo express services
- gradle build files
- basic spring boot service (an hello world application)
- logback configuration (with configured ECS json encoder)

#### Motivation and Context
Those modifications are needed in order to setup project

#### How Has This Been Tested?
- Local gradle build
- Local docker build
- Local docker compose build

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+version: '3.8'
+
+services:
+  pagopa-ecommerce-user-service:
```
Comment (scaminati-bv) [inline docker-compose.yml:4]:
```suggestion
  pagopa-ecommerce-user-stats-service:
```

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    container_name: pagopa-ecommerce-user-service
```
Comment (scaminati-bv) [inline docker-compose.yml:5]:
```suggestion
    container_name: pagopa-ecommerce-user-stats-service
```

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      - ./.env
+    ports:
+      - "8080:8080"
+    depends_on:
+      - mongo
+    networks:
+      - pagopa-ecommerce-user-net
+
+  mongo-express:
+    container_name: pagopa-ecommerce-user-service-mongo-express
```
Comment (scaminati-bv) [inline docker-compose.yml:19]:
```suggestion
    container_name: pagopa-ecommerce-user-stats-service-mongo-express
```

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    env_file: ./docker/config/mongo/mongo-express.env
+    ports:
+      - "8081:8081"
+    networks:
+      - pagopa-ecommerce-user-net
+    depends_on:
+      - mongo
+
+  mongo:
+    container_name: pagopa-ecommerce-user-service-mongo
```
Comment (scaminati-bv) [inline docker-compose.yml:30]:
```suggestion
    container_name: pagopa-ecommerce-user-stats-service-mongo
```

File: docker/config/mongo/mongo-express.env [hunk 5]
```diff
@@ -0,0 +1,3 @@
+MONGO_INITDB_ROOT_USERNAME=admin
+MONGO_INITDB_ROOT_PASSWORD=password
+ME_CONFIG_MONGODB_URL=mongodb://$MONGO_INITDB_ROOT_USERNAME:$MONGO_INITDB_ROOT_PASSWORD@pagopa-ecommerce-user-service-mongo:27017/
```
Comment (scaminati-bv) [inline docker/config/mongo/mongo-express.env:3]:
```suggestion
ME_CONFIG_MONGODB_URL=mongodb://$MONGO_INITDB_ROOT_USERNAME:$MONGO_INITDB_ROOT_PASSWORD@pagopa-ecommerce-user-stats-service-mongo:27017/
```

File: settings.gradle [hunk 6]
```diff
@@ -0,0 +1,10 @@
+pluginManagement {
+repositories {
+	gradlePluginPortal()
+}
+}
+/*
+ * This file was generated by the Gradle 'init' task.
+ */
+
+rootProject.name = 'pagopa-ecommerce-user-service'
```
Comment (scaminati-bv) [inline settings.gradle:10]:
```suggestion
rootProject.name = 'pagopa-ecommerce-user-stats-service'
```

File: build.gradle.kts [hunk 7]
```diff
@@ -0,0 +1,209 @@
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+group = "it.pagopa.ecommerce.users"
+
+version = "0.0.1"
+
+description = "pagopa-ecommerce-user-service"
```
[the remark this answers is not in the document]
Comment (pietro-tota) [reply build.gradle.kts]: Fixed with https://github.com/pagopa/pagopa-ecommerce-user-stats-service/pull/1/commits/4c070ad9c63fc53fa1968d5f8db83ce0480bf850

File: docker-compose.yml (hunk 1 above)
Comment (pietro-tota) [reply docker-compose.yml]: Fixed with https://github.com/pagopa/pagopa-ecommerce-user-stats-service/pull/1/commits/4c070ad9c63fc53fa1968d5f8db83ce0480bf850

File: docker-compose.yml (hunk 3 above)
Comment (pietro-tota) [reply docker-compose.yml]: Fixed with https://github.com/pagopa/pagopa-ecommerce-user-stats-service/pull/1/commits/4c070ad9c63fc53fa1968d5f8db83ce0480bf850

File: docker/config/mongo/mongo-express.env (hunk 5 above)
Comment (pietro-tota) [reply docker/config/mongo/mongo-express.env]: Fixed with https://github.com/pagopa/pagopa-ecommerce-user-stats-service/pull/1/commits/4c070ad9c63fc53fa1968d5f8db83ce0480bf850

File: settings.gradle (hunk 6 above)
Comment (pietro-tota) [reply settings.gradle]: Fixed with https://github.com/pagopa/pagopa-ecommerce-user-stats-service/pull/1/commits/4c070ad9c63fc53fa1968d5f8db83ce0480bf850

File: docker-compose.yml (hunk 2 above)
Comment (pietro-tota) [reply docker-compose.yml]: Fixed with https://github.com/pagopa/pagopa-ecommerce-user-stats-service/pull/1/commits/4c070ad9c63fc53fa1968d5f8db83ce0480bf850

File: docker-compose.yml (hunk 4 above)
Comment (pietro-tota) [reply docker-compose.yml]: Fixed with https://github.com/pagopa/pagopa-ecommerce-user-stats-service/pull/1/commits/4c070ad9c63fc53fa1968d5f8db83ce0480bf850

File: README.md [hunk 8]
```diff
@@ -1,2 +1,259 @@
-# pagopa-ecommerce-user-service
-Service responsabile of handling user informations such as last used payment method during a transaction on eCommerce
+# PagoPA eCommerce user stats service
+
+[![Quality Gate Status](https://sonarcloud.io/api/project_badges/measure?project=pagopa_pagopa-ecommerce-user-stats-service&metric=alert_status)](https://sonarcloud.io/dashboard?id=pagopa_pagopa-wallet-service)
```
Comment (pietro-tota) [inline README.md:3]:
```suggestion
[![Quality Gate Status](https://sonarcloud.io/api/project_badges/measure?project=pagopa_pagopa-ecommerce-user-stats-service&metric=alert_status)](https://sonarcloud.io/dashboard?id=pagopa_pagopa-ecommerce-user-stats-service)
```

**Review by scaminati-bv — DISMISSED** [review]

File: README.md [hunk 9]
```diff
@@ -1,2 +1,259 @@
...
+- jdk-17
+
+### Run the project
+
+```shell
+$ export $(grep -v '^#' .env.local | xargs)
+$ ./gradlew bootRun
+```
+
+### Install eCommerce commons library locally
```
Comment (ciuffagianluca) [inline README.md:95]: remove this section

File: README.md [hunk 10]
```diff
@@ -1,2 +1,259 @@
...
+
+---
+
+## Develop Locally 💻
+
+### Prerequisites
+
+- git
+- gradle
+- jdk-17
```
Comment (scaminati-bv) [inline README.md:86]:
```suggestion
- jdk-21
```

**Comment by ciuffagianluca** [comment]
Just some trivial comment to resolve, then pr is ok

File: README.md (hunk 9 above)
Comment (pietro-tota) [reply README.md]: fixed with https://github.com/pagopa/pagopa-ecommerce-user-stats-service/pull/1/commits/302f276de3f8eecff1fe3504e50e54c7bdc4546c

File: README.md (hunk 10 above)
Comment (pietro-tota) [reply README.md]: Fixed with https://github.com/pagopa/pagopa-ecommerce-user-stats-service/pull/1/commits/302f276de3f8eecff1fe3504e50e54c7bdc4546c

**Review by scaminati-bv — APPROVED** [review]

**Pull request merged**